app/app.go

```diff
@@ -172,19 +173,10 @@ func (app *PlasmaMVPChain) endBlocker(ctx sdk.Context, req abci.RequestEndBlock)
 	var header [32]byte
 	copy(header[:], ctx.BlockHeader().DataHash)
 	block := plasma.NewBlock(header, app.txIndex, app.feeAmount)
-	plasmaBlockNum := app.plasmaStore.StoreBlock(ctx, tmBlockHeight, block)
-	app.ethConnection.CommitPlasmaHeaders(ctx, app.plasmaStore)
-
-	if app.feeAmount.Sign() != 0 {
-		// add a utxo in the store with position 2^16-1
-		utxo := store.UTXO{
-			Position: plasma.NewPosition(plasmaBlockNum, 1<<16-1, 0, nil),
-			Output:   plasma.NewOutput(app.operatorAddress, app.feeAmount),
-			Spent:    false,
-		}
-
-		app.utxoStore.StoreUTXO(ctx, utxo)
-	}
+	app.blockStore.StoreBlock(ctx, tmBlockHeight, block)
+	// TODO verify
+	app.outputStore.StoreFee(ctx, plasma.NewPosition(app.blockStore.PlasmaBlockHeight(ctx), 1<<16-1, 0, big.NewInt(0)), plasma.NewOutput(app.operatorAddress, app.feeAmount))
```

Comment:
Should a Fee be created if `feeAmount == 0`?

Reply:
no it shouldn't, we should add that check